CMakeLists.txt

```diff
@@ -54,19 +54,29 @@ else ()
 endif ()
 
 
+# Even in "Debug" build type, static linking prevents storing some symbol information and makes debugging more difficult.
+# use this flag to avoid `$1 = <incomplete type>` messages
+if (NOT KEEP_SYMBOL_INFO)
+  set (KEEP_SYMBOL_INFO false CACHE BOOL "Even in Debug build type, static linking prevents storing some symbol information and makes debugging more difficult. Use this flag to avoid `$1 = <incomplete type>` messages" FORCE)
+endif ()
+
 # Compiler and linkers flags
 if (LINUX)
-  # Static Build Extra Flags for Linux Static Build
-  set (BUILD_SHARED_LIBS OFF)
-  set (CMAKE_EXE_LINKER_FLAGS "-static")
-  set (CMAKE_FIND_LIBRARY_SUFFIXES .a)
-  set (CMAKE_EXE_LINK_DYNAMIC_C_FLAGS)       # remove -Wl,-Bdynamic
-  set (CMAKE_EXE_LINK_DYNAMIC_CXX_FLAGS)
-  set (CMAKE_SHARED_LIBRARY_C_FLAGS)         # remove -fPIC
-  set (CMAKE_SHARED_LIBRARY_CXX_FLAGS)
-  set (CMAKE_SHARED_LIBRARY_LINK_C_FLAGS)    # remove -rdynamic
-  set (CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS)
   set (Boost_USE_STATIC_LIBS ON)             # only find static libs
+  if (NOT KEEP_SYMBOL_INFO)
```

Comment:
would it be cool to say `if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")`

Reply:
the thing is I think the problem is linking the system libs statically. if we use the condition you suggest, it won't be possible to have a Debug binary with static linking.

Reply:
do we need `debug binary using static linking` ? as you suggested that static linking is creating some problem, so why shouldn't we drop that case `debug + static linking`.

Reply:
I did it this way to keep it independent, and also being able to compare debug+dynamic and debug+static. Not really sure what's the best here